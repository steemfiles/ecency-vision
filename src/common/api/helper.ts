import defaults from "../constants/site.json";

export const apiBase = (endpoint: string): string =>
  `${defaults.base}${endpoint}`;

export const apiBaseImage = (endpoint: string): string =>
  `${defaults.imageServer}${endpoint}`;
