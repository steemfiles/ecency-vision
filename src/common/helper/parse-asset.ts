export enum Symbol {
    HIVE = 'HIVE',
    HBD = 'HBD',
    TESTS = 'TESTS',
    TBD = 'TBD',
}

export interface Asset {
    amount: number,
    symbol: Symbol
}

export default (strVal: string): Asset => {

    const sp = strVal.split(' ');
    return {
        amount: parseFloat(sp[0]),
        symbol: Symbol[sp[1]]
    }
}
