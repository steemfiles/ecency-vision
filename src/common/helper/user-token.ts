import {User} from "../store/users/types";

import {decodeObj} from "../util/encoder";

import * as ls from "../util/local-storage";

export const getUser = (username: string): User => {
    const raw = ls.get(`user_${username}`);
    if (!raw) {
        throw "User does not exist!";
    }

    try {
        return decodeObj(raw) as User;
    } catch (e) {
        throw "User does not exist!";
    }
}

export const getAccessToken = (username: string): string => getUser(username).accessToken;

export const getPostingKey = (username: string): null | undefined | string => getUser(username).postingKey;

export const getRefreshToken = (username: string): string => getUser(username).refreshToken;

