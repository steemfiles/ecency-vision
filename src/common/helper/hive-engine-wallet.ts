import {Account} from "../store/accounts/types";
import {DynamicProps} from "../store/dynamic-props/types";
import HiveWallet from "./hive-wallet";
import {FullHiveEngineAccount, TokenBalance} from "../api/hive-engine";

export default class HiveEngineWallet extends HiveWallet {

    public engineBalanceTable: { [symbol: string]: TokenBalance } = {};
    public engineBalanceArray: Array<TokenBalance> = [];
    public token_unstakes: any = [];
    public token_statuses: any = [];
    public transfer_history: any = [];
    public token_delegations: Array<any> = [];
    public prices : {[id:string]: number} = {};
    public estimatedValue : number = 0;


    constructor(account: Account, dynamicProps: DynamicProps, convertingHBD: number = 0, shortCoinName: string) {
        super(account, dynamicProps, convertingHBD);
        const {hivePerMVests, base, quote} = dynamicProps;
        const pricePerHive = base / quote;
        
        if (!account.__loaded) {
            return;
        }

        {
            const faccount : FullHiveEngineAccount = account as FullHiveEngineAccount;
            if (!faccount.token_balances) {
                return;
            }
            for (const tb of faccount.token_balances) {
                this.engineBalanceTable[tb.symbol] = tb;
            }
            this.engineBalanceArray = faccount.token_balances;
            this.prices = faccount.prices;
            this.token_unstakes = faccount.token_unstakes;
            this.token_statuses = faccount.token_statuses;
            this.transfer_history = faccount.transfer_history;
            this.token_delegations = faccount.token_delegations;
            
            this.estimatedValue = 0;
            const balance = faccount.token_balances.find(B => B.symbol === shortCoinName);
            let p : number | undefined;
            if (balance && (p = faccount.prices[balance.symbol])) {
                const liquid : number = (balance.balance);
                const stake  : number = (balance.stake);
                // delegationsIn is not part of stake, so there is no need to subtract
                //const delegationsIn = parseFloat(balance.delegationsIn);

                // stake doesn't include delegationsOut.  This must be added to the total.
                // pending* attributes are about the future we don't need to add or subtract those for this.
                const delegationsOut : number = balance.delegationsOut;
                this.estimatedValue = (liquid+stake+delegationsOut) * p * pricePerHive;
            }
        }
    }
}
