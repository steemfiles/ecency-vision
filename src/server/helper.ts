import axios, { AxiosResponse, Method } from "axios";
import { baseApiRequest } from "./util";

import { Entry } from "../common/store/entries/types";
import { DynamicProps } from "../common/store/dynamic-props/types";

import { initialState as dynamicPropsInitialState } from "../common/store/dynamic-props";
import * as hiveApi from "../common/api/hive";

import { cache } from "./cache";
import config from "../config";

export const optimizeEntries = (entries: Entry[]): Entry[] => {
  return entries;
  /* Optimization disabled for now
    return entries.map((x) => {
        return {
            ...x,
            ...{active_votes: []}, // remove active votes
        };
    }); */
};

const makeApiAuth = () => {
  try {
    const auth = Buffer.from(config.privateApiAuth, "base64").toString("utf-8");
    return JSON.parse(auth);
  } catch (e) {
    return null;
  }
};

export const apiRequest = (
  endpoint: string,
  method: Method,
  extraHeaders: any = {},
  payload: any = {}
): Promise<AxiosResponse> | Promise<any> => {
  const apiAuth = makeApiAuth();
  if (!apiAuth) {
    return new Promise((resolve, reject) => {
      console.error("Api auth couldn't be created!");
      reject("Api auth couldn't be created!");
    });
  }

  const url = `${config.privateApiAddr}/${endpoint}`;

  const headers = {
    "Content-Type": "application/json",
    ...makeApiAuth(),
    ...extraHeaders,
  };

  return baseApiRequest(url, method, headers, payload);
};

export const fetchPromotedEntries = async (): Promise<Entry[]> => {
  // fetch list from api
  const list: { author: string; permlink: string; post_data?: Entry }[] = (
    await apiRequest("promoted-posts?limit=200", "GET")
  ).data;

  // random sort & random pick 18 (6*3)
  const promoted = list
    .sort(() => Math.random() - 0.5)
    .filter((x, i) => i < 18);

  return promoted.map((x) => x.post_data).filter((x) => x) as Entry[];
};

export const getPromotedEntries = async (): Promise<Entry[]> => {
  let promoted: Entry[] | undefined = cache.get("promoted-entries");
  if (promoted === undefined) {
    try {
      promoted = await fetchPromotedEntries();
      cache.set("promoted-entries", promoted, 60);
    } catch (e) {
      promoted = [];
    }
  }

  return promoted.sort(() => Math.random() - 0.5);
};

export const getSearchIndexCount = async (): Promise<number> => {
  let indexCount: number | undefined = cache.get("index-count");
  if (indexCount === undefined) {
    try {
      indexCount = (await axios
        .get("https://hivesearcher.com/api/count")
        .then((r) => r.data)) as number;
    } catch (e) {
      indexCount = 0;
    }

    cache.set("index-count", indexCount, 86400);
  }

  return indexCount;
};

export const getDynamicProps = async (): Promise<DynamicProps> => {
  let props: DynamicProps | undefined = cache.get("dynamic-props");
  if (props === undefined) {
    try {
      props = await hiveApi.getDynamicProps();
      cache.set("dynamic-props", props, 120);
    } catch (e) {
      props = { ...dynamicPropsInitialState };
    }
  }

  return props;
};
