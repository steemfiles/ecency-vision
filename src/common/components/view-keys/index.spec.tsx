import React from "react";

import TestRenderer from "react-test-renderer";

import ViewKeysDialog, {ViewKeys} from "./index";

import {activeUserMaker} from "../../helper/test-helper";

it("(1) Default render", () => {
    const props = {
        activeUser: activeUserMaker("foo"),

    };
    const renderer = TestRenderer.create(<ViewKeysDialog {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
});


it("(2) Dialog content", () => {
    const props = {
        activeUser: activeUserMaker("foo"),
        onUpdate: () => {
        }
    };
    const renderer = TestRenderer.create(<ViewKeys {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
});
