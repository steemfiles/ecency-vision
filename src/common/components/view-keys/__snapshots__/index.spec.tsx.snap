// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<button
  className="view-keys-btn btn btn-primary btn-sm"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <svg
    fill="currentColor"
    viewBox="0 0 16 16"
  >
    <path
      d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"
    />
    <path
      d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"
    />
  </svg>
   
  View Private keys
</button>
`;

exports[`(2) Dialog content 1`] = `
<div
  className="dialog-content"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="account-name"
      >
        Account
      </label>
      <input
        className="form-control"
        id="account-name"
        readOnly={true}
        type="text"
        value="foo"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="cur-pass"
      >
        Current password
      </label>
      <input
        autoComplete="off"
        autoFocus={true}
        className="form-control"
        id="cur-pass"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <div>
        <button
          className="btn btn-outline-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Click to view keys
        </button>
      </div>
    </div>
  </form>
</div>
`;
